### PR TITLE
MML Characters

### LINKED ISSUE #96: m-character Support

### PR DESCRIPTION
**Pull request by TheCodeTherapy**
Resolves #96 

This PR implements loading of MML Characters to the 3d-web-experience

**What kind of change does your PR introduce?** (check at least one)

- [ ] Bugfix
- [X] Feature
- [ ] Refactor
- [ ] Tests
- [ ] Other, please describe:

**Does your PR fulfill the following requirements?**

- [ ] The title references the corresponding issue # (if relevant)

brave_eMOIFjZoKB

### CODE REVIEW

File: packages/3d-web-avatar/src/character/Character.ts [hunk 1]
```diff
@@ -42,6 +85,9 @@ export class Character {
       if (child.type === "SkinnedMesh") {
         (child as SkinnedMesh).castShadow = true;
         (child as SkinnedMesh).receiveShadow = true;
+        const asSkinnedMesh = child as SkinnedMesh;
+        const asMeshStandardMaterial = asSkinnedMesh.material as MeshStandardMaterial;
+        // asMeshStandardMaterial.envMapIntensity = 2.0;
```
[the remark this answers is not in the document]
Comment (TheCodeTherapy) [reply packages/3d-web-avatar/src/character/Character.ts]: fixed

File: packages/3d-web-avatar/src/helpers/SkeletonHelpers.ts [hunk 2]
```diff
@@ -149,7 +155,7 @@ export class SkeletonHelpers {
         if (hierarchyCheck === false) {
           hierarchyCheck = true;
           this.extractAndStoreBoneHierarchy(node, modelName);
-          this.compareLatestHierarchies();
+          // this.compareLatestHierarchies();
```
[the remark this answers is not in the document]
Comment (TheCodeTherapy) [reply packages/3d-web-avatar/src/helpers/SkeletonHelpers.ts]: fixed

File: packages/3d-web-avatar/src/character/Character.ts [hunk 3]
```diff
@@ -54,15 +100,18 @@ export class Character {
...
-          (child as SkinnedMesh).bind(this.sharedSkeleton!, this.sharedMatrixWorld!);
-          this.skinnedMeshesParent?.children.splice(3, 0, child as SkinnedMesh);
+        const asSkinnedMesh = child as SkinnedMesh;
+        if (asSkinnedMesh.type === "SkinnedMesh") {
+          const skinnedMeshClone = child.clone(true) as SkinnedMesh;
+          this.remapBoneIndices(skinnedMeshClone);
+          skinnedMeshClone.castShadow = true;
+          skinnedMeshClone.receiveShadow = true;
+          skinnedMeshClone.bind(this.sharedSkeleton!, this.sharedMatrixWorld!);
+          this.skinnedMeshesParent?.children.splice(3, 0, skinnedMeshClone as SkinnedMesh);
```
[the remark this answers is not in the document]
Comment (TheCodeTherapy) [reply packages/3d-web-avatar/src/character/Character.ts]: fixed

**Review by MarcusLongmuir — COMMENTED** [review]
This PR could also remove the previous models and animation files to clean up:

```
unreal-air.glb
unreal-idle.glb
unreal-jog.glb
unreal-mesh.glb
unreal-run.glb
```

**Review by MarcusLongmuir — APPROVED** [review]

**Pull request merged**